cylc/flow/task_pool.py

```diff
@@ -1002,53 +1008,106 @@ def report_unmet_deps(self):
                 )
             )
 
+    def hold_active_task(self, itask: TaskProxy) -> None:
+        if itask.state.reset(is_held=True):
+            self.data_store_mgr.delta_task_held(itask)
+        self.tasks_to_hold.add((itask.tdef.name, itask.point))
```

Comment:
Active tasks don't need to go into the `tasks_to_hold` set, if they were in the set before I think they should come out?

Reply:
They don't need to, but I think it's easier to deal with if everything that is held or is going to be held is in the set. They should only need to come out when released.